### DISCUSSION TITLE
graphql-codegen fails with "Error: command failed"

### DISCUSSION THREAD

**Discussion by nareshbhatia** [dotansimha/graphql-code-generator#9918] [Q&A]
Hi, I am trying to use the typescript plugin `@graphql-codegen/typescript` to generate Typescript types from my graphql schema. However the cli is erroring out without much information. Can someone please help?

Here's my codegen configuration:

```ts
// ----- codegen.ts -----
import type { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  schema: './src/gql/**.graphql',
  generates: {
    './src/generated/gql/': {
      plugins: ['typescript'],
      config: {
        verbose: true,
        scalars: {
          DateTime: 'string',
        },
      },
    },
  },
  hooks: { afterAllFileWrite: ['prettier --write'] },
};

export default config;
```

These are the dev dependencies in my package.json:

```json
  "devDependencies": {
    "@graphql-codegen/cli": "^5.0.2",
    "@graphql-codegen/typescript": "^4.0.6",
  }
```

I have boiled down my graphql schema to just one type:

```graphql
type Market {
  id: ID!
  name: String!
}
```

Here's the error output:

```
> graphql-codegen --config codegen.ts

✔ Parse Configuration
✔ Generate outputs
npm ERR! Lifecycle script `_codegen` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: cruise-ref-app-data@1.0.0
npm ERR!   at location: /Users/naresh.bhatia/projects/market-app
ERROR: "_codegen" exited with 1.
```

**Comment by nareshbhatia** [comment]
Still struggling with this one. Can anyone please help? Is the TypeScript plugin standalone, or do I need to add something else? All I am trying to do is to ingest a schema and generate typescript types. Specifically just `type` declarations, not even `input`, `query` or `mutation` declarations.

**Comment by Hexatare** [comment]
I realize this question was asked quite a while ago, but for anyone finding this via Google, I found the solution:

The issue in the provided code example (at least in my case) was that the output was defined as a folder, which doesn't work. You need to specify a single file instead. For example:

`./src/generated/gql/index.ts`

Hope this helps someone!
